GoogleForCreators/web-stories-wp
# Improved error handling when adding links to elements

## Summary

Reduces the chances of a user seeing an error message when trying to add links, even though the URL might be perfectly valid.

## Relevant Technical Choices

Removed almost all instances where we throw an error.

Just because we cannot read a site's metadata doesn't mean we cannot add a link.

## To-do



## User-facing changes

- Lack of error message when the error is likely temporary or access to a site is blocked (e.g. 403 status like on google.com)
- Error message below input field when entering a clearly invalid URL
- Metadata fields are hidden if the URL is invalid

## Testing Instructions

1. Enter `https://www.google.com` or a non-existent URL when adding a link to an element.
1. Verify that nothing happens, no error message is added, and no link image or title is added.
1. Enter `"'"'` when adding a link to an element.
1. Verify that an error message is displayed below the element.
1. Verify that fields for link description & icon are not displayed in case of an error.

---



Fixes #1966
Fixes #983
